From 6c78b427f62aa419aa431ad4273c445788ac4105 Mon Sep 17 00:00:00 2001
From: Michel Pollet <michel.pollet@bp.renesas.com>
Date: Fri, 27 May 2016 08:51:41 +0100
Subject: [PATCH] net: Add a top level NET_VENDOR_RENESAS

Since both the Synopsys GMACs (STMMAC) and the MoreThanIP 5-port
switch create net devices (ethX), we have a problem with probe
ordering. We want the Synopsys GMACs to get eth0 and eth1, so this
driver has to probe first.

Second, we have a run-time dependency that the Renesas RGMII/GMII
Converters must be probed after the MoreThanIP 5-port switch. This
is because the RGMII/GMII Converters need to hook into PHYs that have
already been probed.

This change does that. It's not nice, I don't like it, but here we are.

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
Signed-off-by: Michel Pollet <michel.pollet@bp.renesas.com>
---
 drivers/net/ethernet/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/Makefile b/drivers/net/ethernet/Makefile
index a09423d..f2d7405 100644
--- a/drivers/net/ethernet/Makefile
+++ b/drivers/net/ethernet/Makefile
@@ -67,7 +67,6 @@ obj-$(CONFIG_NET_VENDOR_PASEMI) += pasemi/
 obj-$(CONFIG_NET_VENDOR_QLOGIC) += qlogic/
 obj-$(CONFIG_NET_VENDOR_QUALCOMM) += qualcomm/
 obj-$(CONFIG_NET_VENDOR_REALTEK) += realtek/
-obj-$(CONFIG_NET_VENDOR_RENESAS) += renesas/
 obj-$(CONFIG_NET_VENDOR_RDC) += rdc/
 obj-$(CONFIG_NET_VENDOR_ROCKER) += rocker/
 obj-$(CONFIG_NET_VENDOR_SAMSUNG) += samsung/
@@ -89,3 +88,4 @@ obj-$(CONFIG_NET_VENDOR_VIA) += via/
 obj-$(CONFIG_NET_VENDOR_WIZNET) += wiznet/
 obj-$(CONFIG_NET_VENDOR_XILINX) += xilinx/
 obj-$(CONFIG_NET_VENDOR_XIRCOM) += xircom/
+obj-$(CONFIG_NET_VENDOR_RENESAS) += renesas/
-- 
2.7.4

